### PR TITLE
Add optional chaining

### PR DESCRIPTION
**Pull request by rickycodes**
Based on a discussion we had around a certain error that I saw in Sentry

### CODE REVIEW

File: app/scripts/metamask-controller.js [hunk 1]
```diff
@@ -257,7 +257,7 @@ export default class MetamaskController extends EventEmitter {
           const metamaskState = await this.getState()
           backEndMetaMetricsEvent(metamaskState, {
             customVariables: {
-              errorMessage: txMeta.simulationFails.reason,
+              errorMessage: txMeta.simulationFails?.reason || 'Unknown Reason',
```
Comment (Gudahtt) [inline app/scripts/metamask-controller.js]:
```suggestion
              errorMessage: txMeta.simulationFails?.reason,
```
Is the `Unknown Reason` part required? Falling back to `null` seems perfectly fine - that's what was intended originally I suspect.

File: babel.config.js [hunk 2]
```diff
@@ -19,6 +19,7 @@ module.exports = function (api) {
       '@babel/plugin-transform-runtime',
       '@babel/plugin-proposal-class-properties',
       '@babel/plugin-proposal-object-rest-spread',
+      '@babel/plugin-proposal-optional-chaining'
```
Comment (Gudahtt) [inline babel.config.js]:
```suggestion
      '@babel/plugin-proposal-optional-chaining',
```
This should fix the lint failure

**Review by Gudahtt — DISMISSED** [review]
LGTM!

File: package.json [hunk 3]
```diff
@@ -60,6 +60,7 @@
   },
   "dependencies": {
     "3box": "^1.10.2",
+    "@babel/plugin-proposal-optional-chaining": "^7.8.3",
```
Comment (whymarrh) [inline package.json]: Can we move this into `devDependencies`?

**Review by whymarrh — CHANGES_REQUESTED** [review]

**Comment by whymarrh** [comment]
Rebasing this on the latest `develop` should help with some of the CI errors here

**Review by Gudahtt — APPROVED** [review]

**Review by whymarrh — APPROVED** [review]

**Pull request merged**